### DISCUSSION TITLE
Protocol Name - Trust Registry Query Protocol?

### DISCUSSION THREAD

**Discussion by darrellodonnell** [trustoverip/tswg-trust-registry-protocol#20] [Ideas]
Would the name "Trust Registry Query Protocol" make more sense than "Trust Registry Protocol". 

I raise this as I always getting questions from that want to understand how the TRP helps them add or update registry entries. That's a brutally hard thing to consider, while getting an answer to a simple question is, well, simple:

> Does `Entity X` have `Authorization Y`, in the context of `Ecosystem Governance Framework Z`?

**Comment by trbouma** [comment]
Hello @darrellodonnell  checking to see if this works....

The statement, IMHO above is the essence of the specification. It might be generalized further which I am now musing on.

**Comment by darrellodonnell** [comment]
we should get this statement into the Scope to hit the reader early with the basic idea.

**Reply by trbouma** [reply]
I think it can further normalized to:

`Z` authorizes `X` for action(s) `Y` 

where:

 `X`,`Z` are `verifiable identifiers` bound to `Entity X` and `Entity Z`

and 

`Y` is the set of actions related to authorities/statuses/privileges granted to `X` in the context defined by `Z`

**Comment by darrellodonnell** [comment]
to your point, it is `authorization` or `status` ? that changes the question a lot

**Reply by trbouma** [reply]
I'd suggest sticking with `authorization` - that's what is really about in the end.

**Comment by trbouma** [comment]
Recasting Darrell's statement:
 
```Does Entity X have Authorization Y, in the context of Ecosystem Governance Framework Z?```

into something that is expressed more mathematically and can be implemented:

```
[X]<----(X,Y)SIGNEDBY(Z)-----[Z]
```
The tuple `(X,Y)` means that `X` is authorized for `Y` as signed by `Z`, which is the same as the context of `Z`

What is exactly `Y` and under what conditions is `(X,Y)` signed is specific to the the governance of `Z`.

This is just a generalized case of what happens in DNSSEC with a DS record. For DNSSEC it is just for delegation of signing keys within the context of a hierarchy. In this case it's between identifiers `X` and `Z`, and it's a graph (web) as opposed to a hierarchy'. The  `(X,Y)SIGNEDBY(Z)` defines the authorization relationship.

**Comment by talltree** [comment]
Closing this discussion, since we renamed the specification as suggested 6+ months ago.
